cloudweatherreport/run.py

```diff
@@ -146,35 +152,108 @@ def load_report(self, datastore, index, test_plan):
+                cpu_limit=cpu_limit,
+                credentials_name=os.getenv('{}')
+            )
+        except (CloudNotSupported, UnknownCloudName,
+                UnknownCredentialName) as e:
+            logging.error(
+                'Skipping cloud resource check: {}'.format(str(e)))
+            return None
+
     def run_plan(self, test_plan):
         env = connect_juju_client(self.controller, logging=logging)
         if not env:
             logging.error("Jujuclient could not connect to {} ".format(
                 self.controller))
             return False
         env.name = self.controller
-        provider = (env.info().get("provider-type") or
-                    env.info().get("ProviderType"))
+        env_info = env.info()
+        provider = (env_info.get("provider-type") or
+                    env_info.get("ProviderType"))
         env.provider_name = get_provider_name(provider)
+
         logging.info('Running test on {}.'.format(env.provider_name))
-        try:
-            test_result = self.run_tests(test_plan, env)
-            benchmark_results = self.run_benchmarks(test_plan, env)
-        except Exception:
-            tb = traceback.format_exc()
-            error = "Exception ({}):\n{}".format(env.name, tb)
-            logging.error(error)
-            # create a fake SuiteResult to hold exception traceback
+        resource_available = self.check_cloud_resource(test_plan, env_info)
+        if resource_available is False:
             test_result = model.SuiteResult(
                 provider=env.provider_name,
-                test_outcome='Error Running Tests',
+                test_outcome='INFRA',
                 tests=[model.TestResult(
-                    name='Error Running Tests',
+                    name='Resource not available',
                     duration=0.0,
-                    output=error,
+                    output='Resource not available',
                     result='INFRA',
+                    suite='Resource not available'
                 )])
-            return False
+            benchmark_results = []
+        else:
+            try:
+                test_result = self.run_tests(test_plan, env)
+                benchmark_results = self.run_benchmarks(test_plan, env)
+            except Exception:
+                tb = traceback.format_exc()
+                error = "Exception ({}):\n{}".format(env.name, tb)
+                logging.error(error)
+                # create a fake SuiteResult to hold exception traceback
+                test_result = model.SuiteResult(
+                    provider=env.provider_name,
+                    test_outcome='Error Running Tests',
+                    tests=[model.TestResult(
+                        name='Error Running Tests',
+                        duration=0.0,
+                        output=error,
+                        result='INFRA',
+                    )])
+                return False
```

Comment:
This `return False` isn't actually part of your change but I just noticed that this is entirely skipping writing the report.  We should ensure that the report gets written with the INFRA result.  Same for the controller connection error at the top of this method as well.  If you don't mind fixing that in this PR, great, but we can do it in another issue instead if you'd prefer.